### PR TITLE
feat(DTUPSTREAM-192): Animation PP logo top and text bottom

### PR DESCRIPTION
**Pull request by lucasfrotam**
### Description

This PR is about the Upfront Presentment project and adds a new animation to our experiments.

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues

- https://engineering.paypalcorp.com/jira/browse/DTUPSTREAM-192

### Screenshots

Single loop: 

https://user-images.githubusercontent.com/25916871/145880900-a3d83963-b0a5-42a4-86bc-3783935e134e.mov

Infinite:

https://user-images.githubusercontent.com/25916871/145880231-5c75656e-b75f-4381-b739-f1b2cbeff9f8.mov

❤️  Thank you!

### CODE REVIEW

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+import type {  ContentOptions } from './types';
+
+type AnimationProps = {|
+    paypalButtonContainerElement : Object,
+    animationConfig : Object,
+    designContainer : Object
+|};
+
+export const LOGO_TOP_TEXT_BOTTOM_CONFIG = {
```
Comment (erick2014) [inline src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:17]: hey @lucasfrotam , you can create a constants file and import this constant in here, we already did that in a previous pr: https://github.com/paypal/paypal-checkout-components/pull/1807/files#diff-8c1e1f81a9e134254530315fef6dc4fe05cfe18f95572bb96771b86c65c14c40R8, the idea is to use the same constant for all the animations we can

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    if (paypalLogoElement && paypalButtonContainerElement) {
+        animationConfig = animationContainerWidth < smallMax ?
+            { paypalLogoTopPosition: 25, labelTextFontSize: 7 } : { paypalLogoTopPosition: 30, labelTextFontSize: 9 };
+    } else {
+        return null;
+    }
+   
+    return {
+        designContainer,
+        paypalButtonContainerElement,
```
Comment (erick2014) [inline src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:86]: there is no need to create another object, you can just destructure this object in here, in order to just return the properties you need

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            text-align: center;
+            width: 100%;
+            color: #142C8E;
+            font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
+            position: absolute;
+            top: 25%;
+        }
+        @keyframes reverse-logo {
+            0%,33%{
+                top: -${ animationConfig.paypalLogoTopPosition }%;
```
Comment (erick2014) [inline src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:117]: there is no need to have this object, just return the properties in line 86

**Review by erick2014 — APPROVED** [review]

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    // get the label container that animation will be applied to
+    const paypalButtonContainerElement = designContainer.querySelector(`.${ PAYPAL_BUTTON }`) || null;
+    // get starting position for element so it doesn't jump when animation begins
+    const paypalLogoElement = (paypalButtonContainerElement && paypalButtonContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+
+    // get correct configuration according to the button size (small or medium)
+    if (paypalLogoElement && paypalButtonContainerElement) {
+        animationConfig = animationContainerWidth < smallMax ?
```
Comment (ssono) [inline src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:78]: Can you use this instead for readability?
`animationConfig = animationContainerWidth < smallMax 
    ? { paypalLogoTopPosition: 25, labelTextFontSize: 7 }
    : { paypalLogoTopPosition: 30, labelTextFontSize: 9 };`

**Review by ssono — COMMENTED** [review]
Mostly looks good will approve when build passes

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx (hunk 2 above)
Comment (lucasfrotam) [reply src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:86]: the object has two properties, `paypalLogoTopPosition and labelTextFontSize`, if I do that, we are changing one line to two lines and adding more variables in the next function which will destructure this object. I don't know if this is the best way to clean the code

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx (hunk 4 above)
Comment (lucasfrotam) [reply src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx]: done

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx (hunk 1 above)
Comment (lucasfrotam) [reply src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx]: done

File: src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx (hunk 2 above)
Comment (erick2014) [reply src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx:73]: well we just have two properties so I dont see the need for that, but I think that is personal way to do it. I am okay with what you decide.

**Review by mnicpt — CHANGES_REQUESTED** [review]
No more changes in checkout in this way as per agreement pre-moratorium

**Pull request closed without merging**